sourcemash/categorize.py

```diff
@@ -1,23 +1,101 @@
+    "should", "show", "side", "since", "sincere", "six", "sixty", "so", "some", \
+    "somehow", "someone", "something", "sometime", "sometimes", "somewhere", "still", \
+    "such", "system", "take", "ten", "than", "that", "the", "their", "them", \
+    "themselves", "then", "thence", "there", "thereafter", "thereby", "therefore", \
+    "therein", "thereupon", "these", "they", "thickv", "thin", "third", "this", \
+    "those", "though", "three", "through", "throughout", "thru", "thus", "to", \
+    "together", "too", "top", "toward", "towards", "twelve", "twenty", "two", \
+    "un", "under", "until", "up", "upon", "us", "very", "via", "was", "we", "well", \
+    "were", "what", "whatever", "when", "whence", "whenever", "where", "whereafter", \
+    "whereas", "whereby", "wherein", "whereupon", "wherever", "whether", "which", \
+    "while", "whither", "who", "whoever", "whole", "whom", "whose", "why", "will", \
+    "with", "within", "without", "would", "yet", "you", "your", "yours", "yourself", \
+    "yourselves", "the" ] # Source: http://xpo6.com/list-of-english-stop-words/
 
-	# Give weights to categories found in title
-	for word in title:
-		categories[word] *= 2
+class Categorizer:
 
-	# Return top 2 words in categories
-	most_frequent = categories.most_common(2)
+    def __init__(self):
+        self.title_categories = Counter()
+
+    def categorize_item(self, title, text):
+        cat1 = ""
+        cat2 = ""
+        categories = Counter()
+
+        # Only consider words present in category dictonary
+        title = filter(lambda word: word in self.title_categories, title.split())
+        text = filter(lambda word: word in self.title_categories, text.split())
+        title = [word.strip(punctuation) for word in title]
+        text = [word.strip(punctuation) for word in text]
+
+        # Get word counts from title, text of article
+        categories.update(title + text)
+
+        # Give weights to categories found in title
+        for word in title:
+            categories[word] *= TITLE_WORD_WEIGHT
+
+        # Return top 2 words in categories
+        most_frequent = categories.most_common(2)
+
+        if len(most_frequent) > 0:
+            cat1 = most_frequent[0][0]
+            if len(most_frequent) > 1:
+                cat2 = most_frequent[1][0]
+
+        return cat1, cat2
+
+
+    def reset_title_categories(self):
+        self.title_categories.clear()
+
+        one_day_ago = datetime.now() - timedelta(hours=24)
+        titles = [item.title for item in Item.query.filter(Item.category_1==None, Item.last_updated>=one_day_ago).all()]
+        self.parse_title_categories(titles)
+
+
+    def parse_title_categories(self, titles):
+        for title in titles:
+            for word in title.split():
+                word = word.strip(punctuation)
+                if not self.ignore_word(word):
+                    self.title_categories.update([word])
+
+
+    def ignore_word(self, word):
```

Comment:
Since this is a boolean function, can we rename it to something like is_valid_word?